ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Protocol/Praos.hs

```diff
@@ -48,6 +49,7 @@ data PraosEnvelopeError
   = ObsoleteNode Version Version
   | HeaderSizeTooLarge Natural Natural
   | BlockSizeTooLarge Natural Natural
+  | InvalidCheckpoint -- TODO args
```

Comment:
I suggest `CheckpointMismatch` or `CheckpointConflict` instead of "invalid checkpoint"

Reply:
Renamed to CheckpointMismatch in 8c48082